sqlite: Install pkg-config definitions with -dev package

phi-gamma opened the pull request:
This is a prerequisite to other packages finding sqlite to build
against. Example: libsqlite3-sys in Cargo won’t build on Nixos
without this change.


###### Motivation for this change

pkgconfig should find sqlite3.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

phi-gamma committed:
sqlite: Install pkg-config definitions with -dev package

This is a prerequisite to other packages finding sqlite to build
against.

Signed-off-by: Philipp Gesang <phg@phi-gamma.net>

Mic92 commented:
Please change the target branch of this pull request from `master` to `staging`.

jtojnar commented on pkgs/development/libraries/sqlite/default.nix:
```diff
@@ -56,6 +56,11 @@ stdenv.mkDerivation {
   '';
 
+  postInstall =
+    ''
```
Could you keep the string syntax consistent with the one above?

vcunat commented:
I'm confused, as `./result-dev/lib/pkgconfig/sqlite3.pc` does exist for me before this PR.

phi-gamma commented:
2017-11-27 (Monday), notifications@github.com (Vladimír Čunát):

> I'm confused, as `./result-dev/lib/pkgconfig/sqlite3.pc` does exist for me before this PR.

Now I’m confused too. Yesterday it definitely lacked that file
without the added line which confused the hell out of me in the
first place. I just checked but I can’t reproduce it now …

Must be some mistake on my side. Sorry for wasting everyone’s
time!

vcunat commented:
Maybe it was some older version.

phi-gamma commented:
2017-11-27 (Monday), notifications@github.com (Vladimír Čunát):

> Maybe it was some older version.

I rebased on recent master so that’s possible. Anything that
changed since 17.09 that might have an effect on pkgconfig files?

vcunat commented:
On current 17.09 it has the file as well.
